### PR TITLE
WIP: Data Classifier PostProcessor using ET

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------

The test for HistorySet is missing. 

##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by wangcj05** [comment]
@mandd I need you to create a test with HistorySet.

File: framework/PostProcessors/DataClassifier.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+    """
+    needDict = {}
+    needDict['Functions'] = []
+    for func in self.mapping.values():
+      needDict['Functions'].append(func)
+    return needDict
+
+  def initialize(self, runInfo, inputs, initDict=None):
+    """
+      Method to initialize the pp.
```
Comment (mandd) [inline framework/PostProcessors/DataClassifier.py]: we should spell PP

File: framework/PostProcessors/DataClassifier.py [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+      for key, values in classifierDict['input'].items():
+        calcVal = self.funcDict[key].evaluate("evaluate", tempTargDict)
+        inds, = np.where(values == calcVal)
+        if labelIndex is None:
+           labelIndex = set(inds)
+        else:
+          labelIndex = labelIndex & set(inds)
+      if len(labelIndex) != 1:
+        self.raiseAnError(IOError, "The parameters", ",".join(tempTargDict.keys()), "with values", ",".join([str(el) for el in tempTargDict.values()]), "could not be classifier!")
+      outputDict[self.label] = np.append(outputDict[self.label], classifierDict['output'][self.label][list(labelIndex)[0]])
```
Comment (mandd) [inline framework/PostProcessors/DataClassifier.py]: would it make sense to avoid the append on a np array since we know already the size of outputDict[self.label]?

File: framework/PostProcessors/ETstructure.py [hunk 3]
```diff
@@ -144,7 +144,7 @@ def checkETstructure(self,links,listETs,connectivityMatrix):
         if len(zeroColumns)==0:
             raise IOError('No root ET')
         if len(zeroColumns)==1:
-            rootETID = listETs[zeroColumns]
+            rootETID = listETs[zeroColumns.astype(int)[0]]
```
Comment (mandd) [inline framework/PostProcessors/ETstructure.py]: why is this needed?

File: framework/PostProcessors/DataClassifier.py [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+
+  def __init__(self, messageHandler):
+    """
+      Constructor
+      @ In, messageHandler, MessageHandler, message handler object
+      @ Out, None
+    """
+    PostProcessor.__init__(self, messageHandler)
+    self.printTag   = 'POSTPROCESSOR DataClassifier'
+    self.dynamic    = False
```
[the remark this answers is not in the document]
Comment (wangcj05) [reply framework/PostProcessors/DataClassifier.py]: removed.

File: framework/PostProcessors/DataClassifier.py (hunk 1 above)
Comment (wangcj05) [reply framework/PostProcessors/DataClassifier.py]: done.

File: framework/PostProcessors/DataClassifier.py (hunk 2 above)
Comment (wangcj05) [reply framework/PostProcessors/DataClassifier.py]: fixed.

File: framework/PostProcessors/ETstructure.py (hunk 3 above)
Comment (wangcj05) [reply framework/PostProcessors/ETstructure.py]: This is complained by the new updated libraries.
